elementary/monitor/data_monitoring/data_monitoring_report.py

```diff
@@ -103,22 +103,23 @@ def generate_report(
             models = self.models_api.get_models(exclude_elementary_models)
             sources = self.models_api.get_sources()
             exposures = self.models_api.get_exposures()
+
             models_runs = self.models_api.get_models_runs(
                 days_back=days_back, exclude_elementary_models=exclude_elementary_models
             )
-            tests_metadata = self.tests_api.get_tests_metadata(days_back=days_back)
 
+            test_results_db_rows = self.tests_api.get_all_test_results_db_rows(
```

Comment:
Just an idea, what do you think about having a `TestsAPI.get_tests(...)` that returns, roughly, an object that looks like this?

```py
class TestResults
    results: ...
    runs: ...
```

That way there's only a single call to `TestsAPI` to get test-related data and there's no need to re-normalize the data in `DataMonitoring`. Not very critical though.

Reply:
Yeah I tend to agree that we need to simplify the flow 🙂 
I think that the original idea was to have an "API" call for each screen, but now we can use a single call (specially when both API's uses the same query).
I want to do it as a part of a different PR because I want to add improvements to the json as well (stop duplicating tests metadata on the json for example) which will require changes at the UI as well.